app/src/main/java/com/example/weatherapp/network/WeatherNetworkRepository.kt

```diff
@@ -0,0 +1,73 @@
+package com.example.weatherapp.network
+
+import android.util.Log
+import android.widget.Toast
+import androidx.lifecycle.MutableLiveData
+import com.example.weatherapp.WeatherModel
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import kotlin.coroutines.coroutineContext
+
+class WeatherNetworkRepository : NetworkRepositoryInterface {
```

Comment:
pass the retrofit interface as parameter NetworkApiInterface

Reply:
done